index.php

```diff
@@ -142,6 +142,12 @@ function get_q_value($mime, $accept)
         error_page('Verification Failed', 'Given Code Was Invalid');
     }
 
+    $response = array('me' => USER_URL);
+
+    if ($scope = (explode(':', $code, 3))[2]) {
```

Comment:
`$scope` isn’t compared against anything here, which makes for a falsey comparison in PHP. This means things like a literal `0`-string will be seen as no scope.

Why anyone would request such a scope, I don’t know, but it would still be valid. Instead the `if` is probably better off as:

```php
if ($scope !== '') {
```